src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -18,12 +19,29 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         size_t hashedArgs{};
         if (const auto& args{ actionAndArgs.Args() })
         {
+            // Args are defined, so hash them
             hashedArgs = gsl::narrow_cast<size_t>(args.Hash());
         }
         else
         {
-            std::hash<IActionArgs> argsHash;
-            hashedArgs = argsHash(nullptr);
+            // Args are not defined.
+            // Check if the ShortcutAction supports args.
+            switch (actionAndArgs.Action())
+            {
+#define ON_ALL_ACTIONS_WITH_ARGS(action)                                    \
+    case ShortcutAction::action:                                            \
+        /* If it does, hash the default values for the args.*/              \
+        hashedArgs = gsl::narrow_cast<size_t>(make<action##Args>().Hash()); \

```

Comment:
I low-key hate that this requires _constructing an entire object filled with default values_ to calculate a hash; it could be wildly expensive. Can you profile to make sure we're not making a bunch of objects just to destroy them moments later? Both CPU and memory-wise.

Reply:
I feel like we could get around this by maintaining a static .Empty singleton of each of these if this is what we need to Hash... or cache the hash of the empty one as a singleton...

Reply:
To be fair though none of the ActionArgs constructors are complex at the moment.

As such I "propose" the following solution:
* add `constexpr` to the default constructors of all ActionArgs classes
* add `constexpr` to all their `Hash()` member functions
* write:
  ```cpp
  hashedArgs = implementation::action##Args{}.Hash();
  ```

This will probably compile to constants in release mode and in debug mode it won't allocate anything on the heap.

Reply:
That will be illegal because we're not using `winrt::make` for something that's known to be a _runtimeclass_.